core-bundle/src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -523,9 +523,13 @@ public static function toXhtml($strString)
 	 * @param string $strString The XHTML string
 	 *
 	 * @return string The HTML5 string
+	 *
+	 * @deprecated Deprecated since Contao 4.13, to be removed in Contao 5.0
 	 */
 	public static function toHtml5($strString)
 	{
+		trigger_deprecation('contao/core-bundle', '4.13', 'The "StringUtil::toHtml5()" method has been deprecated and will no longer work in Contao 5.0.');
```

Comment:
```suggestion
		if (str_starts_with(debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2)[1]['class'] ?? '', 'Contao\\')) 
		{
			trigger_deprecation('contao/core-bundle', '4.13', 'The "StringUtil::toHtml5()" method has been deprecated and will no longer work in Contao 5.0.');
		}
```
To not throw self deprecations?

Reply:
Triggering a deprecation is meant for things that people can change to get rid of the deprecation, however, in this case we have to continue using the method, so we should only add a `@deprecated` comment and no `trigger_deprecation()` at all in my opinion.

Reply:
I’d be fine with that too. But wouldn’t this make it harder to notice for everyone using `StringUtil::toHtml5()` themself?
With my suggestion from above, extension developers would get deprecation notices while the core itself does’t trigger them.

Reply:
Agreed.

But why do we not provide a migration for the core fields and stop using the method ourselves?

Reply:
A migration would be hard to get right IMO (and hard to detect if it should be executed).

But a migration that only replaces the ` rel="lightbox` part might be suitable. The lightbox replacement is the only one necessary I think, the others are only “cosmetic” IMO.
The detection would then simly need to check for `LIKE '% rel="lightbox%'`

Reply:
I wouldn't want a migration changing any of my database fields tbh. And on top of detecting the content of the database field we would need to make sure the code in question is actually used (e.g. `$GLOBALS['TL_CTE']['texts']['text']` not overwritten by anyone and so on). 

Then, I'd rather favor a cut in Contao 5 and the :fox_face:-deprecation without changing usages in 4.13.

Reply:
Judging by https://github.com/contao/core/issues/4073 the `rel` attribute was set by tinyMCE. The lightbox option seems gone in the current demo, but maybe this is something that can be configured in there? (producing the wanted tag in the first place)

Reply:
AFAIK `rel="lightbox…` does no longer get inserted by tinyMCE.